### PR TITLE
NTP-771: Adding DfeAnalytics to get analytics data streaming to BigQuery

### PR DESCRIPTION
**Pull request by ashward**
## Context

This adds https://github.com/DFE-Digital/dfe-analytics-dotnet so that anonymised web request data is sent to BigQuery for further analysis.

See: https://github.com/DFE-Digital/dfe-analytics-dotnet and https://github.com/DFE-Digital/dfe-analytics

## Changes proposed in this pull request

- Added `dfe-analytics-dotnet`
- Added the ability to configure it with credentials from the filesystem (rather than an env var)
- Added a check so that the service will still start even if `dfe-analytics-dotnet` is not properly configured

## Link to Jira ticket

https://dfedigital.atlassian.net/browse/NTP-771

## Things to check

- [x] Title is in the format "NTP-XXX: Short description" where NTP-XXX is the Jira ticket reference
- [ ] Code and tests follow the [coding standards](/docs/coding-standards.md)
- [x] `dotnet format` has been run in the repository root
- [ ] Test coverage of new code is at least 80%
- [ ] **PR deployment has been signed off by wider team**

### CODE REVIEW

File: Infrastructure/Analytics/AnalyticsConfigurerService.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+        private bool IsAnalyticsConfigured()
+        {
+            var section = _configuration.GetSection("DfeAnalytics");
+
+            // Do we have the bare minimum of configuration?
+            return (
+                (
+                    !string.IsNullOrEmpty(section["CredentialsJson"])
+                    || (!string.IsNullOrEmpty(section["CredentialsJsonFile"]) && !string.IsNullOrEmpty(section["ProjectId"]))
```
Comment (kieranohanlon) [inline Infrastructure/Analytics/AnalyticsConfigurerService.cs:48]: Since using an OR, I'm not sure it will ever check the Project Id in this clause?

Comment (ashward) [reply Infrastructure/Analytics/AnalyticsConfigurerService.cs:48]: I think the logic is correct, but I agree it's not very obvious what the logic is supposed to be doing! I'll tidy it up a bit.

Comment (ashward) [reply Infrastructure/Analytics/AnalyticsConfigurerService.cs]: Have rearranged the logic a bit.

**Review by kieranohanlon — APPROVED** [review]
Looks good

**Pull request merged**